sdk/core/core/inc/az_span.h

```diff
@@ -14,24 +15,12 @@
 
 #include <_az_cfg_prefix.h>
 
-/**
```

Comment:
I really, really don't like moving the struct definition away from the functions that operate on it, it's really still public and all you're doing is spreading the public API out more.

Besides span is supposed to be a simple aggregate type that we expect users to initialize with designated initializers... right? Something like az_contract seems fine to try and hide in this way, but span not so much.